File: backend/enums/Emotions.py
from enum import Enum


# Enum of the possible emotions that can be found in one text post.
# Based on the Plutchik's wheel of emotions.
class Emotions(Enum):
    ANGER = "anger"
    ANTICIPATION = "anticipation"
    DISGUST = "disgust"
    FEAR = "fear"
    JOY = "joy"
    SADNESS = "sadness"
    SURPRISE = "surprise"
    TRUST = "trust"


# Enum with the midpoint for each pair of emotions on the wheel
class EmotionsMiddle(Enum):
    ANGER_DISGUST = "anger-disgust"
    ANTICIPATION_ANGER = "anticipation-anger"
    DISGUST_SADNESS = "disgust-sadness"
    FEAR_TRUST = "fear-trust"
    JOY_ANTICIPATION = "joy-anticipation"
    SADNESS_SURPRISE = "sadness-surprise"
    SURPRISE_FEAR = "surprise-fear"
    TRUST_JOY = "trust-joy"

File: frontend/src/app/utils/emotions.ts
export class Emotions {
    public ANGER = "anger";
    public ANTICIPATION = "anticipation";
    public DISGUST = "disgust";
    public FEAR = "fear";
    public JOY = "joy";
    public SADNESS = "sadness";
    public SURPRISE = "surprise";
    public TRUST = "trust";

    public EMOTIONS = [
        this.ANGER,
        this.ANTICIPATION,
        this.DISGUST,
        this.FEAR,
        this.JOY,
        this.SADNESS,
        this.SURPRISE,
        this.TRUST
    ];

    constructor() {}

    getColors(emotion: String) {
        if (emotion == this.ANGER) {
            return "rgb(204,51,51)";
        }
        if (emotion == this.ANTICIPATION) {
            return "rgb(204,102,51)";
        }
        if (emotion == this.DISGUST) {
            return "rgb(102,51,153)";
        }
        if (emotion == this.FEAR) {
            return "rgb(51,153,102)";
        }
        if (emotion == this.JOY) {
            return "rgb(255,204,102)";
        }
        if (emotion == this.SADNESS) {
            return "rgb(0,51,204)";
        }
        if (emotion == this.SURPRISE) {
            return "rgb(153,204,255)";
        }
        if (emotion == this.TRUST) {
            return "rgb(153,204,102)";
        }
    }
}
